src/app/InteractionModelEngine.h

```diff
@@ -136,6 +136,7 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
 
 #if CHIP_CONFIG_ENABLE_ICD_SERVER
     void SetICDManager(ICDManager * manager) { mICDManager = manager; };
+    ICDManager * GetICDManager() { return mICDManager; };
```

Comment:
We made a design choice to not use the Icdmanager object outside of the state machine to reduce dependency and avoid friend class that are not directly tied to the ICD itself.

I think we could do what you are doing using ICD Notifier and creating an event to send a check-in message with the right subjectDescriptor 

@mkardous-silabs what are your thoughts on  @yunhanw-google implementation here and on my suggestion.